### PR TITLE
Add separate demo for mulisampling

### PR DESCRIPTION
**Pull request by MrMinimal**
Because the multisampling setting can break the demos on somem systems, this creates a demo dedicated to this feature.

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/671)

### CODE REVIEW

**Comment by Xaeroxe** [comment]
I'm mostly curious what system was broken by that.

**Comment by Rhuagh** [comment]
I think this is a case where we should investigate if we can handle this more gracefully inside the engine, and downgrade the multisampling if the target device don't support it.

File: examples/sphere_multisample/main.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
+//! Displays a shaded sphere to the user.
```
Comment (jojolepro) [inline examples/sphere_multisample/main.rs]: shaded sphere with multisampling enabled

File: examples/sphere_multisample/resources/display_config.ron [hunk 2]
```diff
@@ -0,0 +1,10 @@
+(
+  dimensions: None,
+  max_dimensions: None,
+  min_dimensions: None,
+  fullscreen: false,
+  multisampling: 1,
+  title: "Sphere example",
```
Comment (jojolepro) [inline examples/sphere_multisample/resources/display_config.ron]: Wrong title

File: examples/sphere_multisample/resources/display_config.ron [hunk 3]
```diff
@@ -0,0 +1,10 @@
+(
+  dimensions: None,
+  max_dimensions: None,
+  min_dimensions: None,
+  fullscreen: false,
+  multisampling: 1,
```
Comment (jojolepro) [inline examples/sphere_multisample/resources/display_config.ron]: Can you set the multisampling higher? I think we can have values of 2,4,8,16

**Comment by jojolepro** [comment]
Thanks for that! :)

**Comment by MrMinimal** [comment]
@jojolepro Changes made! Thanks for the review

**Comment by derekdreery** [comment]
I agree about the graceful downgrade. I'm currently having a similar problem running pong because `cpal` is panicking. It should return an error so that the example can warn and then run without sound.

**Comment by Rhuagh** [comment]
Seems like the suggested solution is to not write directly to the framebuffer, but write to a render target with multisampling defined and then just blit to the main framebuffer at the end.

I suggest we defer that change until the new renderer is in place.

**Comment by Rhuagh** [comment]
As a side note, Hal does not support multisampling on backbuffers, so the only way to get support there is using the method I described above.

**Comment by Rhuagh** [comment]
Bors r+

**Pull request merged**